commit ac6bc6159dab28bd2f753a7927aff137b01be252
Author: Susan Franklin <sfranklin>
Date:   Mon Mar 8 22:44:59 2010 +0000

    Bug 304942 -  remove deprecated method use in RuntimeInstallJob

1	7	bundles/org.eclipse.equinox.p2.operations/src/org/eclipse/equinox/p2/operations/ProvisioningSession.java
diff --git a/bundles/org.eclipse.equinox.p2.operations/src/org/eclipse/equinox/p2/operations/ProvisioningSession.java b/bundles/org.eclipse.equinox.p2.operations/src/org/eclipse/equinox/p2/operations/ProvisioningSession.java
index ff80f41..7241b30 100644
--- a/bundles/org.eclipse.equinox.p2.operations/src/org/eclipse/equinox/p2/operations/ProvisioningSession.java
+++ b/bundles/org.eclipse.equinox.p2.operations/src/org/eclipse/equinox/p2/operations/ProvisioningSession.java
@@ -84,13 +84,7 @@ public class ProvisioningSession {
 		return (IMetadataRepositoryManager) agent.getService(IMetadataRepositoryManager.SERVICE_NAME);
 	}
 
-	/**
-	 * Return the profile registry for this session
-	 * @return the profile registry
-	 * 
-	 * @deprecated this API will not appear in the final release.
-	 */
-	public IProfileRegistry getProfileRegistry() {
+	IProfileRegistry getProfileRegistry() {
 		return (IProfileRegistry) agent.getService(IProfileRegistry.SERVICE_NAME);
 	}
 
commit 50f813b9e67526689b6e01a899b0fad229479806
Author: Susan Franklin <sfranklin>
Date:   Mon Mar 8 23:05:58 2010 +0000

    Bug 304942 -  remove deprecated method use in RuntimeInstallJob

2	4	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/QueryProvider.java
1	1	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/actions/ProfileModificationAction.java
2	3	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/model/ProfileElement.java
2	3	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/model/ProfileSnapshots.java
1	1	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/model/RollbackProfileElement.java
2	1	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/query/QueryableProfileRegistry.java
1	1	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/p2/ui/RevertProfilePage.java
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/QueryProvider.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/QueryProvider.java
index 7a3d58d..4f10b46 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/QueryProvider.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/QueryProvider.java
@@ -11,8 +11,6 @@
  *******************************************************************************/
 package org.eclipse.equinox.internal.p2.ui;
 
-import org.eclipse.equinox.p2.query.QueryUtil;
-
 import java.net.URI;
 import java.util.Collection;
 import org.eclipse.equinox.internal.p2.ui.model.*;
@@ -75,7 +73,7 @@ public class QueryProvider {
 				IProfile targetProfile = null;
 				String profileId = context.getInstalledProfileId();
 				if (profileId != null) {
-					targetProfile = ui.getSession().getProfileRegistry().getProfile(profileId);
+					targetProfile = ProvUI.getProfileRegistry(ui.getSession()).getProfile(profileId);
 				}
 
 				IQuery<IInstallableUnit> topLevelQuery = policy.getVisibleAvailableIUQuery();
@@ -126,7 +124,7 @@ public class QueryProvider {
 				IProfile profile;
 				IInstallableUnit[] toUpdate = null;
 				if (element instanceof Updates) {
-					profile = ui.getSession().getProfileRegistry().getProfile(((Updates) element).getProfileId());
+					profile = ProvUI.getProfileRegistry(ui.getSession()).getProfile(((Updates) element).getProfileId());
 					toUpdate = ((Updates) element).getIUs();
 				} else {
 					profile = ProvUI.getAdapter(element, IProfile.class);
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/actions/ProfileModificationAction.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/actions/ProfileModificationAction.java
index e9b309d..8231c4b 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/actions/ProfileModificationAction.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/actions/ProfileModificationAction.java
@@ -58,7 +58,7 @@ public abstract class ProfileModificationAction extends ProvisioningAction {
 
 	public IProfile getProfile() {
 		String id = profileId == null ? ui.getProfileId() : profileId;
-		return ui.getSession().getProfileRegistry().getProfile(id);
+		return ProvUI.getProfileRegistry(ui.getSession()).getProfile(id);
 	}
 
 	protected IStatus getNoProfileOrSelectionStatus(String id, Collection<IInstallableUnit> ius) {
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/model/ProfileElement.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/model/ProfileElement.java
index 2da1928..35a17d3 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/model/ProfileElement.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/model/ProfileElement.java
@@ -10,8 +10,7 @@
  *******************************************************************************/
 package org.eclipse.equinox.internal.p2.ui.model;
 
-import org.eclipse.equinox.internal.p2.ui.ProvUIImages;
-import org.eclipse.equinox.internal.p2.ui.QueryProvider;
+import org.eclipse.equinox.internal.p2.ui.*;
 import org.eclipse.equinox.p2.engine.IProfile;
 import org.eclipse.equinox.p2.query.IQueryable;
 
@@ -53,7 +52,7 @@ public class ProfileElement extends RemoteQueriedElement {
 	}
 
 	public IQueryable<?> getQueryable() {
-		return getProvisioningUI().getSession().getProfileRegistry().getProfile(profileId);
+		return ProvUI.getProfileRegistry(getProvisioningUI().getSession()).getProfile(profileId);
 	}
 
 	/*
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/model/ProfileSnapshots.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/model/ProfileSnapshots.java
index d4b10e1..12a24b5 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/model/ProfileSnapshots.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/model/ProfileSnapshots.java
@@ -12,8 +12,7 @@ package org.eclipse.equinox.internal.p2.ui.model;
 
 import org.eclipse.core.runtime.IProgressMonitor;
 import org.eclipse.core.runtime.jobs.ISchedulingRule;
-import org.eclipse.equinox.internal.p2.ui.ProvUIActivator;
-import org.eclipse.equinox.internal.p2.ui.ProvUIMessages;
+import org.eclipse.equinox.internal.p2.ui.*;
 import org.eclipse.ui.progress.IDeferredWorkbenchAdapter;
 import org.eclipse.ui.progress.IElementCollector;
 
@@ -39,7 +38,7 @@ public class ProfileSnapshots extends ProvElement implements IDeferredWorkbenchA
 	 * @see org.eclipse.ui.model.IWorkbenchAdapter#getChildren(java.lang.Object)
 	 */
 	public Object[] getChildren(Object o) {
-		long[] timestamps = ProvUIActivator.getDefault().getSession().getProfileRegistry().listProfileTimestamps(profileId);
+		long[] timestamps = ProvUI.getProfileRegistry(ProvUIActivator.getDefault().getSession()).listProfileTimestamps(profileId);
 		RollbackProfileElement[] elements = new RollbackProfileElement[timestamps.length];
 		boolean skipFirst = false;
 		for (int i = 0; i < timestamps.length; i++) {
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/model/RollbackProfileElement.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/model/RollbackProfileElement.java
index 5258326..d474a73 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/model/RollbackProfileElement.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/model/RollbackProfileElement.java
@@ -60,7 +60,7 @@ public class RollbackProfileElement extends RemoteQueriedElement {
 
 	public IProfile getProfileSnapshot(IProgressMonitor monitor) {
 		if (snapshot == null) {
-			snapshot = getProvisioningUI().getSession().getProfileRegistry().getProfile(profileId, timestamp);
+			snapshot = ProvUI.getProfileRegistry(getProvisioningUI().getSession()).getProfile(profileId, timestamp);
 			setQueryable(snapshot);
 		}
 		return snapshot;
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/query/QueryableProfileRegistry.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/query/QueryableProfileRegistry.java
index e14b99a..4d787c0 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/query/QueryableProfileRegistry.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/query/QueryableProfileRegistry.java
@@ -14,6 +14,7 @@ package org.eclipse.equinox.internal.p2.ui.query;
 import java.util.Arrays;
 import org.eclipse.core.runtime.IProgressMonitor;
 import org.eclipse.core.runtime.SubMonitor;
+import org.eclipse.equinox.internal.p2.ui.ProvUI;
 import org.eclipse.equinox.internal.p2.ui.ProvUIMessages;
 import org.eclipse.equinox.p2.engine.IProfile;
 import org.eclipse.equinox.p2.query.*;
@@ -31,7 +32,7 @@ public class QueryableProfileRegistry implements IQueryable<IProfile> {
 	}
 
 	public IQueryResult<IProfile> query(IQuery<IProfile> query, IProgressMonitor monitor) {
-		IProfile[] profiles = ui.getSession().getProfileRegistry().getProfiles();
+		IProfile[] profiles = ProvUI.getProfileRegistry(ui.getSession()).getProfiles();
 		SubMonitor sub = SubMonitor.convert(monitor, ProvUIMessages.QueryableProfileRegistry_QueryProfileProgress, profiles.length);
 		try {
 			return query.perform(Arrays.asList(profiles).iterator());
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/p2/ui/RevertProfilePage.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/p2/ui/RevertProfilePage.java
index 7dc1bb3..feefb02 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/p2/ui/RevertProfilePage.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/p2/ui/RevertProfilePage.java
@@ -543,7 +543,7 @@ public class RevertProfilePage extends InstallationPage implements ICopyable {
 				// If it is a recognized element and it is not the current profile, then it can be deleted.
 				if (selected instanceof RollbackProfileElement && !((RollbackProfileElement) selected).isCurrentProfile()) {
 					RollbackProfileElement snapshot = (RollbackProfileElement) selected;
-					IProfileRegistry registry = getSession().getProfileRegistry();
+					IProfileRegistry registry = ProvUI.getProfileRegistry(getSession());
 					if (registry != null) {
 						try {
 							registry.removeProfile(profileId, snapshot.getTimestamp());
commit 1d4913bf89a5f8289f1f375936de4e8c907b453a
Author: Susan Franklin <sfranklin>
Date:   Mon Mar 8 23:12:08 2010 +0000

    Bug 304942 -  remove deprecated method use in RuntimeInstallJob

2	3	bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/AddProfileJob.java
2	3	bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProfileFactory.java
1	1	bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProfilesView.java
7	0	bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProvAdminUIActivator.java
1	1	bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/RemoveProfilesJob.java
2	2	bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/dialogs/IUProfilePropertiesGroup.java
diff --git a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/AddProfileJob.java b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/AddProfileJob.java
index 4a3efe4..ecc86f0 100644
--- a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/AddProfileJob.java
+++ b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/AddProfileJob.java
@@ -10,10 +10,9 @@
  *******************************************************************************/
 package org.eclipse.equinox.internal.p2.ui.admin;
 
-import org.eclipse.equinox.p2.core.ProvisionException;
-
 import java.util.Map;
 import org.eclipse.core.runtime.*;
+import org.eclipse.equinox.p2.core.ProvisionException;
 import org.eclipse.equinox.p2.operations.ProvisioningJob;
 import org.eclipse.equinox.p2.operations.ProvisioningSession;
 
@@ -35,7 +34,7 @@ public class AddProfileJob extends ProvisioningJob {
 	public IStatus runModal(IProgressMonitor monitor) {
 		IStatus status = Status.OK_STATUS;
 		try {
-			getSession().getProfileRegistry().addProfile(profileId, profileProperties);
+			ProvAdminUIActivator.getDefault().getProfileRegistry().addProfile(profileId, profileProperties);
 		} catch (ProvisionException e) {
 			status = getErrorStatus(null, e);
 		}
diff --git a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProfileFactory.java b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProfileFactory.java
index cae6b67..831213f 100644
--- a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProfileFactory.java
+++ b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProfileFactory.java
@@ -10,13 +10,12 @@
  *******************************************************************************/
 package org.eclipse.equinox.internal.p2.ui.admin;
 
-import org.eclipse.equinox.p2.core.ProvisionException;
-
 import java.util.HashMap;
 import java.util.Map;
 import org.eclipse.core.runtime.Platform;
 import org.eclipse.equinox.internal.p2.core.helpers.ServiceHelper;
 import org.eclipse.equinox.internal.p2.ui.ProvUIActivator;
+import org.eclipse.equinox.p2.core.ProvisionException;
 import org.eclipse.equinox.p2.engine.IProfile;
 import org.eclipse.osgi.service.environment.EnvironmentInfo;
 
@@ -40,7 +39,7 @@ public class ProfileFactory {
 		profileProperties.put(IProfile.PROP_NL, getDefaultNL());
 
 		try {
-			return ProvAdminUIActivator.getDefault().getProvisioningUI(null).getSession().getProfileRegistry().addProfile(profileId, profileProperties);
+			return ProvAdminUIActivator.getDefault().getProfileRegistry().addProfile(profileId, profileProperties);
 		} catch (ProvisionException e) {
 			// log
 		}
diff --git a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProfilesView.java b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProfilesView.java
index df5cef1..7946d12 100644
--- a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProfilesView.java
+++ b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProfilesView.java
@@ -199,7 +199,7 @@ public class ProfilesView extends ProvView {
 	}
 
 	String[] getKnownProfileIds() {
-		IProfile[] allProfiles = getProvisioningUI().getSession().getProfileRegistry().getProfiles();
+		IProfile[] allProfiles = ProvAdminUIActivator.getDefault().getProfileRegistry().getProfiles();
 		String[] ids = new String[allProfiles.length];
 		for (int i = 0; i < allProfiles.length; i++)
 			ids[i] = allProfiles[i].getProfileId();
diff --git a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProvAdminUIActivator.java b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProvAdminUIActivator.java
index a12b2b3..0b93343 100644
--- a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProvAdminUIActivator.java
+++ b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProvAdminUIActivator.java
@@ -13,6 +13,8 @@ package org.eclipse.equinox.internal.p2.ui.admin;
 import org.eclipse.equinox.internal.p2.core.helpers.ServiceHelper;
 import org.eclipse.equinox.internal.p2.ui.ProvUIActivator;
 import org.eclipse.equinox.internal.p2.ui.admin.preferences.PreferenceConstants;
+import org.eclipse.equinox.p2.core.IProvisioningAgent;
+import org.eclipse.equinox.p2.engine.IProfileRegistry;
 import org.eclipse.equinox.p2.engine.query.UserVisibleRootQuery;
 import org.eclipse.equinox.p2.operations.RepositoryTracker;
 import org.eclipse.equinox.p2.query.QueryUtil;
@@ -141,6 +143,11 @@ public class ProvAdminUIActivator extends AbstractUIPlugin {
 		return policy;
 	}
 
+	public IProfileRegistry getProfileRegistry() {
+		IProvisioningAgent agent = (IProvisioningAgent) ServiceHelper.getService(context, IProvisioningAgent.SERVICE_NAME);
+		return (IProfileRegistry) agent.getService(IProfileRegistry.SERVICE_NAME);
+	}
+
 	public ProvisioningUI getProvisioningUI(String profileId) {
 		return new ProvisioningUI(ProvisioningUI.getDefaultUI().getSession(), profileId, policy);
 	}
diff --git a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/RemoveProfilesJob.java b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/RemoveProfilesJob.java
index 94d1eec..b43a74c 100644
--- a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/RemoveProfilesJob.java
+++ b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/RemoveProfilesJob.java
@@ -29,7 +29,7 @@ public class RemoveProfilesJob extends ProvisioningJob {
 
 	public IStatus runModal(IProgressMonitor monitor) {
 		for (int i = 0; i < profileIds.length; i++) {
-			getSession().getProfileRegistry().removeProfile(profileIds[i]);
+			ProvAdminUIActivator.getDefault().getProfileRegistry().removeProfile(profileIds[i]);
 		}
 		return Status.OK_STATUS;
 
diff --git a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/dialogs/IUProfilePropertiesGroup.java b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/dialogs/IUProfilePropertiesGroup.java
index 468b5ed..db4f4c6 100644
--- a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/dialogs/IUProfilePropertiesGroup.java
+++ b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/dialogs/IUProfilePropertiesGroup.java
@@ -10,10 +10,10 @@
  *******************************************************************************/
 package org.eclipse.equinox.internal.p2.ui.admin.dialogs;
 
+import org.eclipse.equinox.internal.p2.ui.admin.ProvAdminUIActivator;
 import org.eclipse.equinox.internal.p2.ui.admin.ProvAdminUIMessages;
 import org.eclipse.equinox.internal.p2.ui.model.InstalledIUElement;
 import org.eclipse.equinox.p2.engine.IProfile;
-import org.eclipse.equinox.p2.ui.ProvisioningUI;
 import org.eclipse.swt.SWT;
 import org.eclipse.swt.events.ModifyListener;
 import org.eclipse.swt.layout.GridData;
@@ -75,6 +75,6 @@ public class IUProfilePropertiesGroup extends IUGroup {
 	}
 
 	private IProfile getProfile(InstalledIUElement element) {
-		return ProvisioningUI.getDefaultUI().getSession().getProfileRegistry().getProfile(element.getProfileId());
+		return ProvAdminUIActivator.getDefault().getProfileRegistry().getProfile(element.getProfileId());
 	}
 }
